src/Day05.kt

```diff
@@ -2,7 +2,7 @@ fun main() {
     fun part1(input: List<String>): Int {
 
         // Extension functions to check for each of the criteria
-        fun String.hasUnwantedStrings() = this.zipWithNext().any { it.first == 'a' && it.second == 'b' } ||
+        fun String.hasUnwantedStrings() = this.windowed(2).any { it.first() == 'a' && it.last() == 'b' } ||
                 this.zipWithNext().any { it.first == 'c' && it.second == 'd' } ||
```

Comment:
I swapped `zipWithNext` here with `windowed` just to demonstrate how `windowed` works in comparison, but then I saw you actually used it below... :D